### PR TITLE
add filter to video dashboard

### PR DESCRIPTION
**Pull request by eacevedo79**
added a filter input

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -5,10 +5,12 @@
     "dependencies": {
         "@novnc/novnc": "^1.1.0",
         "ansi-256-colors": "^1.1.0",
+        "build": "^0.1.4",
```
Comment (lanwen) [inline ui/package.json]: what's it for?

File: ui/src/components/Videos/index.js [hunk 2]
```diff
@@ -1,48 +1,62 @@
...
+        this.state = {
+            data: [],
+            value: "",
+        };
+    }
+
+    componentWillMount() {
+        const vids = [];
+        this.props.videos.forEach(function(item, index) {
+            vids.push({ name: item });
```
Comment (lanwen) [inline ui/src/components/Videos/index.js]: better to keep not the videos in the state, but filter only

File: ui/src/components/Videos/index.js [hunk 3]
```diff
@@ -1,48 +1,62 @@
...
+        });
 
-export default Videos;
+        this.setState({ data: vids });
+    }
+
+    handleChange = event => {
+        const { value } = event.target;
+        this.setState({ value });
```
Comment (lanwen) [inline ui/src/components/Videos/index.js]: you can use useState hook from React for the state management, that would make the component stateless on its own

**Comment by lanwen** [comment]
Hello, great feature, thx for implementation. Can you add a screenshot on how it looks like? Also adding the ne dependency just for the filtering capability looks like a bit an overhead, can you explain your thoughts on that, why its necessary to use that library?

File: ui/package.json (hunk 1 above)
Comment (eacevedo79) [reply ui/package.json]: it may be a bad merge.. i'll remove.
thanks

File: ui/src/components/Videos/index.js (hunk 3 above)
Comment (eacevedo79) [reply ui/src/components/Videos/index.js]: i followed the example from here:

**Comment by eacevedo79** [comment]
> Hello, great feature, thx for implementation. Can you add a screenshot on how it looks like? Also adding the ne dependency just for the filtering capability looks like a bit an overhead, can you explain your thoughts on that, why its necessary to use that library?

if there is a better lighter way i'm happy to start a conversation about it.
we notice when we have lots of videos is hard to find them by just scrolling around.
initially my idea was to have a reveal as you scroll feature matched with this filter, but i'm not a experienced front end developer, i'm more a backend/ops kinda dev.

thanks

File: ui/src/components/Videos/index.js (hunk 3 above)
Comment (lanwen) [reply ui/src/components/Videos/index.js]: mb I can push to your branch? But you can spot some examples in the code on how the hook works

Comment (eacevedo79) [reply ui/src/components/Videos/index.js]: Sure feel free to change anything you want

**Comment by lanwen** [comment]
Superseded by #262

**Pull request closed without merging**